sdk/core/Azure.Core.TestFramework/src/ManagementInterceptor.cs

```diff
@@ -31,6 +31,9 @@ public void Intercept(IInvocation invocation)
             var type = result.GetType();
             if (type.Name.StartsWith("Task"))
             {
+                if (((Task)result).IsFaulted || ((Task)result).Status == TaskStatus.WaitingForActivation)
```

Comment:
why not use:
``` c#
var resultTask = result as Task;
if (resultTask is null || resultTask.IsFaulted || resultTask.Status == TaskStatus.WaitingForActivation)
```

also, wonder if we can't lift this guard clause out of this if, as we likely want to return from any kind of task that is faulted or inactive.